pkg/kubectl/cmd/set/set_image.go

```diff
@@ -216,38 +216,20 @@ func (o *SetImageOptions) Run() error {
 	allErrs := []error{}
 
 	patches := CalculatePatches(o.Infos, scheme.DefaultJSONEncoder(), func(obj runtime.Object) ([]byte, error) {
-		transformed := false
 		_, err := o.UpdatePodSpecForObject(obj, func(spec *v1.PodSpec) error {
 			for name, image := range o.ContainerImages {
-				var (
-					containerFound bool
-					err            error
-					resolved       string
-				)
-				// Find the container to update, and update its image
-				for i, c := range spec.Containers {
-					if c.Name == name || name == "*" {
-						containerFound = true
-						if len(resolved) == 0 {
-							if resolved, err = o.ResolveImage(image); err != nil {
-								allErrs = append(allErrs, fmt.Errorf("error: unable to resolve image %q for container %q: %v", image, name, err))
-								// Do not loop again if the image resolving failed for wildcard case as we
-								// will report the same error again for the next container.
-								if name == "*" {
-									break
-								}
-								continue
-							}
-						}
-						if spec.Containers[i].Image != resolved {
-							spec.Containers[i].Image = resolved
-							// Perform updates
-							transformed = true
-						}
+				resolvedImageName, err := o.ResolveImage(image)
```

Comment:
Would you mind just removing the `ResolveImage` hack which looks like over-engineering. No-one seem to be overriding this, and it's just an identity function anyway, it can't fail, and this still complicates this code for no reason.

Reply:
In this file 'ResolveImage' not doing anything, just returning image name. But  'ResolveImage' functionality added in this PR https://github.com/kubernetes/kubernetes/pull/33083 . I thought it may be used in some thirdparty code which depends on kubectl package.  Is it ok if it is removed from here?

Reply:
I'd prefer we leave the `ResolveImage` as is, it does allow nicely switching implementation for dependents.